bitswap.go

```diff
@@ -386,6 +396,8 @@ func (bs *Bitswap) receiveBlocksFrom(ctx context.Context, from peer.ID, blks []b
 // ReceiveMessage is called by the network interface when a new message is
 // received.
 func (bs *Bitswap) ReceiveMessage(ctx context.Context, p peer.ID, incoming bsmsg.BitSwapMessage) {
+	now := time.Now()
```

Comment:
nit: I don't think we actually _want_ to be this accurate about the receive time. We're trying to measure how long it takes from "want sent" to "block seen and recorded". Given that, I think we want to count the "received" time where we cancel the "don't have timeouts".

That should also simplify this code as we don't need to thread through the receive time.

Reply:
Good point 👍 I'll simplify